src/transformers/trainer.py

```diff
@@ -3201,12 +3246,24 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
 
                 The dictionary will be unpacked before being fed to the model. Most models expect the targets under the
                 argument `labels`. Check your model's documentation for all accepted arguments.
+            kwargs:
+                Additional key-word arguments to pass along for custom optimizers
```

Comment:
We should try to avoid adding in kwargs as much as possible, particularly when they're not needed. 

```suggestion
```

It's acceptable, although not ideal, to have the creation and preparation of kwargs within the method itself, but we really don't want to suddely allow uses to pass forward whatever they want to `self.accelerator.backward`